modules/database/test/std/rec/addrModifierTest.c

```diff
@@ -0,0 +1,247 @@
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+
+#include <string.h>
+
+#include "dbAccess.h"
+#include "devSup.h"
+#include "alarm.h"
+#include "dbUnitTest.h"
+#include "errlog.h"
+#include "epicsThread.h"
+
+#include "longinRecord.h"
+
+#include "testMain.h"
+
+void recTestIoc_registerRecordDeviceDriver(struct dbBase *);
+
+static void startTestIoc(const char *dbfile)
+{
+    testdbPrepare();
+    testdbReadDatabase("recTestIoc.dbd", NULL, NULL);
+    recTestIoc_registerRecordDeviceDriver(pdbbase);
+    testdbReadDatabase(dbfile, NULL, NULL);
+
+    eltc(0);
+    testIocInitOk();
+    eltc(1);
+}
+
+struct pv {
+    const char *name;
+    long count;
+    double values[4];
+};
+
+static void expectProcSuccess(struct pv *pv)
+{
+    char fieldname[20];
+    testdbPutFieldOk("reset.PROC", DBF_LONG, 1);
+    testDiag("expecting success from %s", pv->name);
+    sprintf(fieldname, "%s.PROC", pv->name);
+    testdbPutFieldOk(fieldname, DBF_LONG, 1);
+    sprintf(fieldname, "%s.SEVR", pv->name);
+    testdbGetFieldEqual(fieldname, DBF_LONG, NO_ALARM);
+    sprintf(fieldname, "%s.STAT", pv->name);
+    testdbGetFieldEqual(fieldname, DBF_LONG, NO_ALARM);
+}
+
+#if 0
+static void expectProcFailure(struct pv *pv)
+{
+    char fieldname[20];
+    testdbPutFieldOk("reset.PROC", DBF_LONG, 1);
+    testDiag("expecting failure S_db_badField %#x from %s", S_db_badField, pv->name);
+    sprintf(fieldname, "%s.PROC", pv->name);
+    testdbPutFieldFail(S_db_badField, fieldname, DBF_LONG, 1);
+    sprintf(fieldname, "%s.SEVR", pv->name);
+    testdbGetFieldEqual(fieldname, DBF_LONG, INVALID_ALARM);
+    sprintf(fieldname, "%s.STAT", pv->name);
+    testdbGetFieldEqual(fieldname, DBF_LONG, LINK_ALARM);
+}
+#endif
+
+static double initial[] = {0,1,2,3,4,5,6,7,8,9};
+static double buf[10];
+
+static void changeRange(struct pv *pv, long start, long incr, long end)
+{
+    char linkstring[60];
+    char pvstring[10];
+
+    if (incr)
+        sprintf(linkstring, "tgt.[%ld:%ld:%ld]", start, incr, end);
+    else if (end)
+        sprintf(linkstring, "tgt.[%ld:%ld]", start, end);
+    else
+        sprintf(linkstring, "tgt.[%ld]", start);
+    testDiag("modifying %s.OUT link: %s", pv->name, linkstring);
+    sprintf(pvstring, "%s.OUT", pv->name);
+    testdbPutFieldOk(pvstring, DBF_STRING, linkstring);
+}
+
+static void expectRange(double *values, long start, long incr, long end)
+{
+    int i,j;
+    if (!incr)
+        incr = 1;
+    for (i=0; i<10; i++)
+        buf[i] = initial[i];
+    for (i=0, j=start; j<=end; i++, j+=incr)
+        buf[j] = values[i];
+    testdbGetFieldEqual("tgt.NORD", DBF_LONG, 8);
+    testdbGetFieldEqual("tgt.SEVR", DBF_LONG, NO_ALARM);
+    testdbGetFieldEqual("tgt.STAT", DBF_LONG, NO_ALARM);
+    testdbGetArrFieldEqual("tgt.VAL", DBF_DOUBLE, 10, 8, buf);
+}
+
+#if 0
```

Comment:
... also

Reply:
> > ... Should I follow the example of ...
> 
> This is no longer necessary for `epicsThread.h` as the libCom headers are decorated with `LIBCOM_API` instead of `epicsShare*`.

I meant the general procedure, not the particular header file. In abf9e68a878 I tried to follow this example. This made some of the windows build problems go away, but I readily admit I have no idea what I am doing here Any advice is welcome. Is there a general guide-line I can study?